### ISSUE TITLE
SOMns does not support Newspeaks setter sends, but uses Smalltalk's `:=` assignment

### ISSUE THREAD

**Issue by smarr** [smarr/SOMns#7]
To be spec. compliant, this should be fixed.

**Comment by richard-roberts** [comment]
So, the message `::=` should invoke a method on the object who owns the variable being assigned to? If so, then we need to first resolve the (potentially implicit) variable to find the correct receiver and then invoke the message with the value of the expression.

Should the object `Object` implement this method?

**Comment by smarr** [comment]
Just for clarification, `var := val` needs to change to `var:: val` everywhere in the code.
In the implementation, it depends on what exactly `var` is what happens, but, all necessary functionality is already there. This is merely a parser change.

For instance, if `var` is a local variable, it will still use a `LocalVariableWriteNode` or `NonLocalVariableWriteNode` (see [Local.getWriteNode()](https://github.com/smarr/SOMns/blob/master/src/som/compiler/Variable.java#L184)).

If var is a slot on an object, it is going to be a `var:` send. And that is already how it works, too.
That's all done in [`getSetterSend()`](https://github.com/smarr/SOMns/blob/master/src/som/compiler/MethodBuilder.java#L460), I think.

For a pull request realizing this, it would be great to have a script that replaces all `SPACE:=` with `::SPACE` in the `.som` files. Probably just some short command-line magic should do the trick. If I am not mistaken, that should be all that's needed for the `.som` file conversion.

**Comment by smarr** [comment]
A simple script to perform the conversion is:

```bash
find . -name '*.som' -exec sed -i.bak 's/\ :=\ /:: /g' {} \;
```

or without backup files:

```bash
find . -name '*.som' -exec sed -i 's/\ :=\ /:: /g' {} \;
```

**Comment by smarr** [comment]
The above transformation is not sufficient for cases where there are multiple whitespaces before `:=`.

To adapt to that, the following needs to be run multiple times:
```
find . -name '*.som' -exec sed -i.bak 's/\ ::\ /::  /g' {} \;
```

It essentially replaces `_::_` with `::__` where `_` is a space in order to ensure that previous alignment for assignments is preserved.

**Comment by smarr** [comment]
Fixed with #170.
